figures/helpers.py

```diff
@@ -188,6 +194,15 @@ def prev_day(val):
     return days_from(val, -1)
 
 
+def utc_yesterday():
+    """Get "yesterday" form the utc datetime
+
+    We primarily use this for the daily metrics collection. However, it proves
+    handy as a convenience function for exploring data in the Django shell
+    """
+    return prev_day(datetime.datetime.utcnow().date())
```

Comment:
My standard advice anytime I see `datetime.now()` or an equivalent, is that you should allow a `now` parameter to be passed in in place of it. Ie,

```python
def utc_yesterday(now=None):
    if not now:
        now = datetime.datetime.utcnow()
    return prev_day(now.date())
```

This makes it considerably easier to test, particularly for the weird edge cases that come up around dates (daylight savings, leap years, etc.) and eliminate the flakey tests that can come up when tests potentially behave differently depending on when they are run (what happens when this code is run exactly at midnight?).

Code that calls `utc_yesterday()` also ought to allow their "now" to be overridden for the same reasons, though they would then only need to pass that along to `utc_yesterday()`. That also tends to make it simpler to implement things like backfill operations, since you can run some code as if it were run at some different time.

Reply:
You raise an excellent point make a good argument. I started to draft a reply, but then the reply started to grow longer. So I moved that off for later follow up so I can focus on getting part 2 in PR and come back to this topic. I just wanted to thank you now for reviewing so soon after I posted the PR before the weekend